### PR TITLE
Update react 16

### PR DESCRIPTION
**Pull request by davgur**

### CODE REVIEW

File: src/components/BaseClasses/ListBase.js [hunk 1]
```diff
@@ -47,6 +47,8 @@ class ListBase extends PureComponent {
 
   render() {
     throw new Error('Not Implemented');
+    // eslint-disable-next-line
```
Comment (edoshor) [inline src/components/BaseClasses/ListBase.js]: eslint disable comment should be inlined and specific error not just disable.

For Example, `var thing = new Thing(); // eslint-disable-line no-use-before-define`

File: src/components/ContentUnit/tabs/associations/CUModal.js [hunk 2]
```diff
@@ -24,12 +24,23 @@ class CUModal extends ListWithCheckboxBase {
...
+      return null;
+    }
+
+    const { wipAssociate, fetchList, handleToggleModal } = props;
+    if (wipAssociate) {
+      return { wip: true };
     }
+
+    if (state.wip && !wipAssociate) {
+      fetchList(NS_UNIT_ASSOCIATION_CU, 1);
```
Comment (edoshor) [inline src/components/ContentUnit/tabs/associations/CUModal.js]: This comments repeats everywhere you changed `componentWillReceiveProps` 

`getDerivedStateFromProps` is not the place for network calls and other side effects.
You are advised to use `componentDidUpdate`

Please read:
https://reactjs.org/docs/react-component.html#static-getderivedstatefromprops
https://reactjs.org/docs/react-component.html#componentdidupdate

And consider each case on its own

File: src/components/BaseClasses/ListBase.js (hunk 1 above)
Comment (davgur) [reply src/components/BaseClasses/ListBase.js]: https://github.com/Bnei-Baruch/mdb-admin/pull/62/commits/cae584ea7f75a7e7e03e28163284786c8215b522

File: src/components/ContentUnit/tabs/associations/CUModal.js (hunk 2 above)
Comment (davgur) [reply src/components/ContentUnit/tabs/associations/CUModal.js]: https://github.com/Bnei-Baruch/mdb-admin/pull/62/commits/07965e1c25758e3192cf86921586d2317037ade1

**Pull request closed without merging**